### ISSUE TITLE
Add Leading Commas Support (C++, Haskell)

### ISSUE THREAD

**Issue by articuluxe** [IvanMalison/multi-line#6]
My compliments on a slick package.  It looks very powerful, although it is not very clear how to configure it.  

On to my question.  Is it possible to configure `multi-line` for a c++ signature such that the parameters take one per line, but the commas lead the parameters, not trail them?

```
void func(
    int a
    , char b
    , double c
    , bool b = false
    );
```

**Comment by colonelpanic8** [comment]
> My compliments on a slick package. It looks very powerful, although it is not very clear how to configure it.

Yeah, configuration is pretty complicated because of how multi-line aims to support so many different languages, indentation styles.

> On to my question. Is it possible to configure multi-line for a c++ signature such that the parameters take one per line, but the commas lead the parameters, not trail them?

Wow, I've never heard of that multi-lining style before. Yes, that should be possible but it will require writing a tiny bit of elisp. Let me see if I can cook something up.

**Comment by colonelpanic8** [comment]
Hah, okay, this is a mouthful, but it was actually pretty easy. Single lining doesn't really work, but I'm pretty sure that I could get it there, but I think that I would need to tweak the way respacing works.

``` emacs-lisp
(defvar multi-line-leading-commas-strategy
  (make-instance 'multi-line-strategy
                 :find
                 (make-instance 'multi-line-forward-sexp-find-strategy
                                :split-advance-fn (lambda ()))
                 :respace
                 (multi-line-default-respacers
                  (multi-line-clearing-reindenting-respacer
                   (make-instance 'multi-line-always-newline)))))
```

EDIT: oh and to make it active in C++ buffers you would add:

``` emacs-lisp
(multi-line-defhook c++ multi-line-leading-commas-strategy)
```

If you want this strategy everywhere:

``` emacs-lisp
(setq-default multi-line-leading-commas-strategy)
```

**Comment by colonelpanic8** [comment]
@danrharms can let me know when you get a chance to test this out so I can close this issue

**Comment by articuluxe** [comment]
Sure will do, sorry I'm away from my computer today.
